File: app/src/main/java/com/wipro/android/proficiencyexercise/domain/repository/RepositoryImpl.java
package com.wipro.android.proficiencyexercise.domain.repository;

import com.wipro.android.proficiencyexercise.model.CanadaList;
import com.wipro.android.proficiencyexercise.netcomm.ServiceAPI;

import javax.inject.Inject;

import io.reactivex.Observable;

public class RepositoryImpl implements Repository {
    ServiceAPI apiInterface;

    @Inject
    public RepositoryImpl(ServiceAPI apiInterface) {
        this.apiInterface = apiInterface;
    }

    @Override
    public Observable<CanadaList> getCanadaList() {
        return apiInterface.getCanadaList();
    }
}

File: app/src/main/java/com/wipro/android/proficiencyexercise/domain/repository/RepositoryImpl.kt
package com.wipro.android.proficiencyexercise.domain.repository

import com.wipro.android.proficiencyexercise.data.remote.api.ServiceAPI
import com.wipro.android.proficiencyexercise.data.remote.response.CanadaList
import io.reactivex.Observable
import javax.inject.Inject

class RepositoryImpl @Inject
constructor(val apiInterface: ServiceAPI) : Repository {

    override val canadaList: Observable<CanadaList>
        get() = apiInterface.canadaList
}
